ITISFoundation/osparc-simcore
# 🐛 Fix file-picker downstream service notification issues

## What do these changes do?



- [x] file-picker propagated updates to downstream services due to a side effect (**FIXED**)
- [x] file-picker properly sends notifications when the file source changes
- [x] dynamic-sidecar properly purges directory when it downloads new data
- [x] dynamic-sidecar inputs downloads are now queued and can no longer be happening in parallel


## Related issue/s



- https://github.com/ITISFoundation/osparc-issues/issues/622

## How to test



1. create a jupyter-math:2.0.5
2. connect a file-picker
3. add a file to the file-picker
4.  see that the contes is displayed in the inputs_1 directory
5.  remove the file form the file-picker (reset button)
6. see that the inputs_1 is directory is empty
7. repeat again from step `3.`

## Checklist
